src/lib/y2users/autoinst/reader.rb

```diff
@@ -98,12 +103,13 @@ def create_password(user)
       # @param user [Hash] a user representation in the format used by UsersSimple
       # @return [String]
       def shadow_string(user)
-        pwd_settings = user["password_settings"] || {}
-        other_attrs = SORTED_SHADOW_ATTRS.map do |attr|
-          pwd_settings[attr] || ""
+        other_attrs = if user.password_settings
+          SORTED_SHADOW_ATTRS.map { |a| user.password_settings.send(a) || "" }
```

Comment:
I wonder if we have here the very same problem that we are addressing with *UsersSimple*. I mean, there is no way to distinguish nil and "" values. For example, a nil "min" means that the value is unknown, so the writer will no keep it as it is. But "" value means that there is no minimum password age.

Reply:
This method returns a String representing the entry in the shadow file, AFAIU (I did not implement this part). And given that the values in `other_attrs` are joined using ':', I guess the `||` is not needed at all. However, I do not think the difference between nil and "" in this context is relevant.

Reply:
We have discussed the issue "offline", as I did not understand the problem. It seems that this approach of ensembling the shadow line does not work when you need to distinguish nil and "". So in UsersSimple, we are following a different approach. We should do the same in another PR, following @joseivanlopez's approach.

Thanks for clarifying!